ironfish/src/rpc/adapters/httpAdapter.ts

```diff
@@ -164,86 +167,123 @@ export class RpcHttpAdapter implements IRpcAdapter {
     response: http.ServerResponse,
     requestId: string,
   ): Promise<void> {
-    if (this.router === null || this.router.server === null) {
-      throw new RpcResponseError('Tried to connect to unmounted adapter')
-    }
-
-    const router = this.router
-
-    if (request.url === undefined) {
-      throw new RpcResponseError('No request url provided')
-    }
+    let chunkStreamed = false
 
-    this.logger.debug(
-      `Call HTTP RPC: ${request.method || 'undefined'} ${request.url || 'undefined'}`,
-    )
+    try {
+      if (this.router === null || this.router.server === null) {
+        throw new RpcResponseError('Tried to connect to unmounted adapter')
+      }
 
-    const route = this.formatRoute(request)
-    if (route === undefined) {
-      throw new RpcResponseError('No route found')
-    }
+      const router = this.router
 
-    // TODO(daniel): clean up reading body code here a bit of possible
-    let size = 0
-    const data: Buffer[] = []
+      if (request.url === undefined) {
+        throw new RpcResponseError('No request url provided')
+      }
 
-    for await (const chunk of request) {
-      Assert.isInstanceOf(chunk, Buffer)
-      size += chunk.byteLength
-      data.push(chunk)
+      this.logger.debug(
+        `Call HTTP RPC: ${request.method || 'undefined'} ${request.url || 'undefined'}`,
+      )
 
-      if (size >= MAX_REQUEST_SIZE) {
-        throw new RpcResponseError('Max request size exceeded')
+      const route = this.formatRoute(request)
+      if (route === undefined) {
+        throw new RpcResponseError('No route found')
       }
-    }
-
-    const combined = Buffer.concat(data)
 
-    this.inboundTraffic.add(size)
-    this.inboundBytes.value += size
+      // TODO(daniel): clean up reading body code here a bit of possible
+      let size = 0
+      const data: Buffer[] = []
 
-    // TODO(daniel): some routes assume that no data will be passed as undefined
-    // so keeping that convention here. Could think of a better way to handle?
-    const body = combined.length ? combined.toString('utf8') : undefined
+      for await (const chunk of request) {
+        Assert.isInstanceOf(chunk, Buffer)
+        size += chunk.byteLength
+        data.push(chunk)
 
-    let chunkStreamed = false
-    const rpcRequest = new RpcRequest(
-      body === undefined ? undefined : JSON.parse(body),
-      route,
-      (status: number, data?: unknown) => {
-        response.statusCode = status
-        const delimeter = chunkStreamed ? MESSAGE_DELIMITER : ''
+        if (size >= MAX_REQUEST_SIZE) {
```

Comment:
Should this be configurable by runner? Wondering why we put arbitrary 5MB limit on this.

Reply:
This is already in the code from the existing HTTP adapter.